lib/SILOptimizer/Mandatory/TFDifferentiation.cpp

```diff
@@ -2225,6 +2225,77 @@ class PrimalGenCloner final : public SILClonerWithScopes<PrimalGenCloner> {
     SILClonerWithScopes::visitReleaseValueInst(rvi);
   }
 
+  void visitStructExtractInst(StructExtractInst *sei) {
+    // Special handling logic only applies when the `struct_extract` is active.
+    // If not, just do standard cloning.
+    if (!activityInfo.isActive(sei, synthesis.indices)) {
+      LLVM_DEBUG(getADDebugStream() << "Not active:\n" << *sei << '\n');
+      SILClonerWithScopes::visitStructExtractInst(sei);
+      return;
+    }
+
+    // This instruction is active. Replace it with a call to the corresponding
+    // getter's VJP.
+
+    // Find the corresponding getter and its VJP.
+    auto *getterDecl = sei->getField()->getGetter();
+    assert(getterDecl);
+    auto *getterFn = getContext().getModule().lookUpFunction(
+        SILDeclRef(getterDecl, SILDeclRef::Kind::Func));
+    if (!getterFn) {
```

Comment:
When will a getter not exist?

Reply:
One example is the case that I added to `test/AutoDiff/autodiff_diagnostics.swift`.

I don't know why the getter doesn't exist in that case, and if I put the same code in a file an compile it with a plain call to `swiftc`, the getter does exist and the code ends up triggering the case where the getter exists but doesn't have a VJP.

I haven't investigated any farther than that, so I don't really know what's going on but the logic seems to handle all the cases I can think of correctly.